lib/elastomer/client/docs.rb

```diff
@@ -265,12 +272,44 @@ def count( query, params = nil )
       #
       # See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-delete-by-query.html
       #
-      # Returns the response body as a hash
+      # Returns a Hash of statistics about the delete operations
       def delete_by_query( query, params = nil )
         query, params = extract_params(query) if params.nil?
 
-        response = client.delete '/{index}{/type}/_query', update_params(params, :body => query, :action => 'docs.delete_by_query')
-        response.body
+        responses = []
+        accumulate = lambda { |response| responses << response unless response == nil }
+
+        accumulate.call(bulk do |b|
+          scan(query, params).each_document do |hit|
+            accumulate.call(b.delete(_id: hit["_id"]))
```

Comment:
Are you sure that `_routing` is required for deletes? This says it's optional: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete.html#delete-routing

Even if we wanted to provide the `_routing` parameter (in order to avoid the delete operation being broadcast to all shards), I don't think we can in general because the routing parameter that was used to index the document is not always stored. It's only retrievable when routing is read from a field in the document, but that's deprecated https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-routing-field.html#_path_2.